backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -60,6 +62,12 @@
     private String currentXmlns;
 
     /**
+     * Provides access to data instances in the evaluation context built from the session.
+     * Maps frame step IDs to data instances; only containing entries in the current frame.
+     */
+    private final Hashtable<String, ExternalDataInstance> instances;
```

Comment:
I'm confused about the key here. Is it the ID of the frame step, or is it the ID of the instance itself? Forcing the two to be the same could lead to a lot of confusion/ambiguity v. just recording which index the step is in the current frame.